staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/equality.go

```diff
@@ -33,42 +34,56 @@ import (
 	"k8s.io/klog/v2"
 )
 
-func determineAvoidNoopTimestampUpdatesEnabled() bool {
-	if avoidNoopTimestampUpdatesString, exists := os.LookupEnv("KUBE_APISERVER_AVOID_NOOP_SSA_TIMESTAMP_UPDATES"); exists {
-		if ret, err := strconv.ParseBool(avoidNoopTimestampUpdatesString); err == nil {
-			return ret
-		} else {
-			klog.Errorf("failed to parse envar KUBE_APISERVER_AVOID_NOOP_SSA_TIMESTAMP_UPDATES: %v", err)
-		}
-	}
-
-	// enabled by default
-	return true
-}
-
 var (
-	avoidNoopTimestampUpdatesEnabled = determineAvoidNoopTimestampUpdatesEnabled()
+	// For each update, the new object is checked to see if it is semantically
+	// changed from the previous object.  If it is semantically unchanged, the
+	// write is ignored.
+	//
+	// Customqualities are functions with the signature:
+	//		func (a, b T) bool
+	//
+	// Customqualities can be used to control how the change comparison is
+	// performed for a type of object, T
+	CustomEqualities             []interface{}
```

Comment:
I thought we were trying to late bind so that custom equality funcs people had already [added into `equality.Semantic.AddFunc(...)`](https://grep.app/search?q=.Semantic.AddFunc) would get picked up in our copy, not so we could expose another list of custom funcs for people to register into

that would allow us to avoid any additional surface area

Reply:
@apelisse I remember you had opinions on this matter

Reply:
Actually not a strong opinion. I don't know how common it is for people to customize their equality function but if they do I agree it should probably be done in `equality.Semantic` so that we automatically grab all their existing changes and doesn't increase the surface area.

It looks like the only problem we have today is that we have to get lucky and to get our code initialized after other people's init() functions are called, then late-binding sounds like a reasonable fix.

Reply:
removed CustomEqualities

lgtm now

Reply:
Yeah, looks good to me too.